src/main/scala/com/sumologic/sumopush/actor/LogProcessor.scala

```diff
@@ -59,36 +59,42 @@ object LogProcessor extends MessageProcessor {
     Behaviors.receiveMessage[LogMessage[_]] {
       case ConsumerLogMessage(record, offset, replyTo) =>
         context.system.log.trace("log key: {}", record.key())
-        val reply = record.value() match {
-          case Success(_@JsonLogEvent(JNothing)) =>
-            context.log.warn("ignoring empty message")
-            (None, offset)
-          case null =>
-            context.log.warn("ignoring null message")
-            (None, offset)
-          case Success(log@JsonLogEvent(_)) =>
-            val requests = createSumoRequestsFromLogEvent(config, record.topic(), log, context.log)
-            if (requests.isEmpty) (None, offset)
-            else {
-              requests.foreach(request => stats.messagesProcessed.labels(request.key.value, request.endpointName).inc())
-              (Some(requests), offset)
-            }
-          case Success(log@KubernetesLogEvent(_, _, _, metadata)) =>
-            val req = if (findContainerExclusion(log)) {
-              stats.messagesIgnored.labels(metadata.container).inc()
-              None
-            } else {
-              val requests = createSumoRequestsFromLogEvent(config, log)
-              requests.foreach(request => stats.messagesProcessed.labels(log.metadata.container, request.endpointName).inc())
-              Some(requests)
-            }
-            (req, offset)
-          case Failure(e) =>
-            val exName = e.toString.split(":")(0)
-            messages_failed.labels(exName).inc()
-            context.log.error("unable to parse log message {}", e.getMessage, e)
+        val reply =  try {
```

Comment:
This is ok but i would like to see this try/catch block moved into another method that returns Try(Seq(SumoRequest)) and is implemented with a Try{ } block (much like the examples here https://stackoverflow.com/questions/17500955/returning-a-value-from-try-catch). Then we can consolidate the handling of unhandled exceptions/null or empty value messages/parse errors etc. into one case block similar to line 87.